hypex/dataset/dataset.py

```diff
@@ -319,6 +320,41 @@ def _check_other_dataset(self, other):
             raise ConcatDataError(type(other))
         if type(other._backend) is not type(self._backend):
             raise ConcatBackendError(type(other._backend), type(self._backend))
+        
+    def astype(
+        self,
+        dtype: Dict[str, DefaultRoleTypes],
```

Comment:
type name should be passed as str or type object, not as an object of one of the DefaultRoleTypes